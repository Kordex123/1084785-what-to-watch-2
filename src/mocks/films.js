const films = [
  {
    id: 10111,
    genre: `Comedies`,
    starring: [],
    movieTitle: `Fantastic Beasts: The Crimes of Grindelwald`,
    previewMovieImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewVideoLink: `http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerMeltdowns.mp4`,
  },
  {
    id: 20222,
    genre: `Comedies`,
    movieTitle: `Bohemian Rhapsody`,
    previewMovieImage: `bohemian-rhapsody.jpg`,
    previewVideoLink: `http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerFun.mp4`,
  },
  {
    id: 30333,
    genre: `Documentary`,
    movieTitle: `Macbeth`,
    previewMovieImage: `macbeth.jpg`,
    previewVideoLink: `https://file-examples.com/wp-content/uploads/2017/04/file_example_MP4_640_3MG.mp4`,
  },
  {
    id: 40444,
    genre: `Romance`,
    movieTitle: `Aviator`,
    previewMovieImage: `aviator.jpg`,
    previewVideoLink: `https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/360/Big_Buck_Bunny_360_10s_1MB.mp4
`,
  },
  {
    id: 50555,
    genre: `Romance`,
    movieTitle: `Pulp Fiction`,
    previewMovieImage: `pulp-fiction.jpg`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  },
];

export default films;
