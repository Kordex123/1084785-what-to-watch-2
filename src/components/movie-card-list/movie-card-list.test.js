import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import MovieCardList from "./movie-card-list";
import films from "../../mocks/films";
import configureStore from "redux-mock-store";
import Namespace from "../../reducer/namespace";

it(`MovieCardList correctly renders after relaunch`, () => {

  const mockStore = configureStore([]);
  let store = mockStore({
    [Namespace.MOVIE]: {
      movieCards: films,
    },
    [Namespace.GENRE]: {
      genre: films[0].genre,
    }
  });

  // eslint-disable-next-line no-console
  console.log(`---------------------`);
  // eslint-disable-next-line no-console
  console.log(store.getState());
  // eslint-disable-next-line no-console
  console.log(`---------------------`);

  const onHover = jest.fn();
  const tree = renderer
    .create(
        <Provider store={store}>
          <MovieCardList onHover={onHover}/>
        </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
