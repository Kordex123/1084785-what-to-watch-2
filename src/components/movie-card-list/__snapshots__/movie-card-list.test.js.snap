// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardList correctly renders after relaunch 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <video
      className="player__video"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      poster="fantastic-beasts-the-crimes-of-grindelwald.jpg"
      type="video/mp4"
    >
      <source
        src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerMeltdowns.mp4"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <video
      className="player__video"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      poster="bohemian-rhapsody.jpg"
      type="video/mp4"
    >
      <source
        src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerFun.mp4"
      />
    </video>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>
</div>
`;
